pkg/action/executor/k8s_controller.go

```diff
@@ -329,3 +339,23 @@ func (c *parentController) GetGitOpsConfig(obj *unstructured.Unstructured) gitop
 	glog.V(3).Infof("No GitOps configuration override found for [%v]. Using default configuration.", appName)
 	return c.gitConfig
 }
+
+// This gets all the resource paths for the containers from the controller spec for the supported controller types
+// resourcePath is similar for Deployment, StatefulSet and DaemonSet - "spec.template.spec.containers[?(@.name==“xxx”)].resources"
+func getResourcePath(controllerType string, k8sSpec *k8sControllerSpec) ([]string, error) {
+	var resourcePaths []string
+	switch controllerType {
+	case util.KindDeployment, util.KindDaemonSet, util.KindStatefulSet:
+		for _, container := range k8sSpec.podSpec.Containers {
+			resourcePath := ".spec.template.spec.containers[?(@.name==" + "\"" + container.Name + "\"" + ")].resources"
+			resourcePaths = append(resourcePaths, resourcePath)
+		}
+		glog.V(4).Infof("found resource paths for supported controller type: %v", controllerType)
+		return resourcePaths, nil
+		// TODO: we don't support any custom controllers currently, since when building podspec it resolves in unexpected controller type.
+		// default resource path for custome custroller - ".spec.containers[?(@.name=="xxxx")].resources"
+	default:
+		glog.V(4).Infof("no resource paths found due to unsupported controller type: %v", controllerType)
+		return resourcePaths, fmt.Errorf("unsupported controller type")
```

Comment:
```go
return resourcePaths, fmt.Errorf("unsupported controller type %v", controllerType)
```

Reply:
will get the default case removed and return empty resourcePaths and error if it doesn't satisfy the switch case. I feel may be keep the log info before returning ?